src/databricks/labs/ucx/source_code/linters/pyspark.py

```diff
@@ -461,14 +470,33 @@ def _visit_call_nodes(cls, tree: Tree) -> Iterable[tuple[Call, NodeNG]]:
 
 
 class SparkSqlPyLinter(_SparkSqlAnalyzer, PythonLinter, Fixer):
+    """Linter for SparkSQL used within PySpark.
+
+    Examples
+    1. Find table name reference in SparkSQL:
+       ``` python
+       spark.sql("SELECT * FROM hive_metastore.schema.table").collect()
+       ```
+    """
 
     def __init__(self, sql_linter: SqlLinter, sql_fixer: Fixer | None):
         self._sql_linter = sql_linter
         self._sql_fixer = sql_fixer
 
+        # This fixer is a wrapper around the SQL fixer for when SQL is used within Python. To uniquely identify this
```

Comment:
Please look at this. This `Linter` is _awkward_ because it is designed as a wrapper around any `SqlLinter` and `Fixer` so that we support linting SQL code called from Python code, therefore not having a consistent diagnostic code, but a "dynamic" diagnostic code based on the `Fixer` that is passed during initialization